From 147dfa64ef635f0c7f3334253ad8e20102fe5526 Mon Sep 17 00:00:00 2001
From: Ronghua Zeng <quic_ronghuaz@quicinc.com>
Date: Tue, 14 May 2024 14:45:03 +0800
Subject: [PATCH] ath9k: add support to query the GPIO pins from DTS

add some code in ath9k driver to query the BT coex GPIO pin
selection information from DTS and use it for BT coex init
later.

Signed-off-by: Ronghua Zeng <quic_ronghuaz@quicinc.com>
---
 drivers/net/wireless/ath/ath9k/ahb.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/drivers/net/wireless/ath/ath9k/ahb.c b/drivers/net/wireless/ath/ath9k/ahb.c
index 93e456b..fee6802 100644
--- a/drivers/net/wireless/ath/ath9k/ahb.c
+++ b/drivers/net/wireless/ath/ath9k/ahb.c
@@ -248,6 +248,9 @@ static int of_ath_ahb_probe(struct platform_device *pdev)
 	const struct of_device_id *match;
 	const struct of_ath_ahb_data *data;
 	u8 led_pin;
+	u8 bt_active_pin;
+	u8 bt_priority_pin;
+	u8 wlan_active_pin;
 
 	match = of_match_device(of_ath_ahb_match, &pdev->dev);
 	data = (const struct of_ath_ahb_data *)match->data;
@@ -259,6 +262,21 @@ static int of_ath_ahb_probe(struct platform_device *pdev)
 	else
 		pdata->led_pin = -1;
 
+	if (!of_property_read_u8(pdev->dev.of_node, "qca,wlan-active-gpios", &wlan_active_pin))
+		pdata->wlan_active_pin = wlan_active_pin;
+	else
+		pdata->wlan_active_pin = -1;
+
+	if (!of_property_read_u8(pdev->dev.of_node, "qca,bt-active-gpios", &bt_active_pin))
+		pdata->bt_active_pin = bt_active_pin;
+	else
+		pdata->bt_active_pin = -1;
+
+	if (!of_property_read_u8(pdev->dev.of_node, "qca,bt-priority-gpios", &bt_priority_pin))
+		pdata->bt_priority_pin = bt_priority_pin;
+	else
+		pdata->bt_priority_pin = -1;
+
 	if (of_property_read_bool(pdev->dev.of_node, "qca,tx-gain-buffalo"))
 		pdata->tx_gain_buffalo = true;
 
-- 
2.7.4

